test/unit/walk-manifest.spec.js

```diff
@@ -0,0 +1,282 @@
+                .get('/redirect.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/var500000/playlist.m3u8`);
+
+            walker(false, '.', TEST_URL + '/test.m3u8', function(err, resources) {
+                // 4 m3u8 and 8 * 3 segments
+                assert.equal(resources.length, 28);
+                resources.forEach(function(item) {
+                    assert(item.uri.includes('.ts') || item.uri.includes('.m3u8'));
+                });
+                done();
+            });
+        });
+
+        it('should return not throw an error if one sub playlist gets a 404', function(done) {
+
+            nock(TEST_URL)
+                .get('/test.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/playlist.m3u8`)
+                .get('/var256000/playlist.m3u8')
+                .reply(404)
+                .get('/var386000/playlist.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/var386000/playlist.m3u8`)
+                .get('/var500000/playlist.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/var500000/playlist.m3u8`);
+
+
+            walker(false, '.', TEST_URL + '/test.m3u8', function(err, resources) {
+                // 3 m3u8 and 8 * 2 segments
+                assert.equal(resources.length, 19);
+                resources.forEach(function(item) {
+                    assert(item.uri.includes('.ts') || item.uri.includes('.m3u8'));
+                    assert(item.uri !== TEST_URL + '/var256000/playlist.m3u8');
+                });
+                done();
+            });
+        });
+
+        it('should return not throw an error if one sub playlist gets a 500', function(done) {
+
+            nock(TEST_URL)
+                .get('/test.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/playlist.m3u8`)
+                .get('/var256000/playlist.m3u8')
+                .reply(500)
+                .get('/var386000/playlist.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/var386000/playlist.m3u8`)
+                .get('/var500000/playlist.m3u8')
+                .replyWithFile(200, `${process.cwd()}/test/resources/with-sub-manifest/var500000/playlist.m3u8`);
+
+
+            walker(false, '.', TEST_URL + '/test.m3u8', function(err, resources) {
+                // 3 m3u8 and 8 * 2 segments
+                assert.equal(resources.length, 19);
+                resources.forEach(function(item) {
+                    assert(item.uri.includes('.ts') || item.uri.includes('.m3u8'));
+                    assert(item.uri !== TEST_URL + '/var256000/playlist.m3u8');
+                });
+                done();
+            });
+        });
+
+        it('should return not throw an error if one sub playlist gets an error', function(done) {
```

Comment:
that seems like it should be configurable behavior. Is this also asserting that the errors can be made visible?